eth/catalyst/api.go

```diff
@@ -756,3 +758,52 @@ func (api *ConsensusAPI) heartbeat() {
 func (api *ConsensusAPI) ExchangeCapabilities([]string) []string {
 	return caps
 }
+
+// GetPayloadBodiesV1 implements engine_getPayloadBodiesByHashV1 which allows for retrieval of a list
+// of block bodies by the engine api.
+func (api *ConsensusAPI) GetPayloadBodiesByHashV1(hashes []common.Hash) []*beacon.ExecutionPayloadBodyV1 {
+	var bodies = make([]*beacon.ExecutionPayloadBodyV1, len(hashes))
+	for i, hash := range hashes {
+		block := api.eth.BlockChain().GetBlockByHash(hash)
+		if block == nil {
+			continue
+		}
+		bodies[i] = getBody(block)
+	}
+	return bodies
+}
+
+// GetPayloadBodiesByRangeV1 implements engine_getPayloadBodiesByRangeV1 which allows for retrieval of a range
+// of block bodies by the engine api.
+func (api *ConsensusAPI) GetPayloadBodiesByRangeV1(start, count uint64) []*beacon.ExecutionPayloadBodyV1 {
+	if api.eth.BlockChain().CurrentBlock().NumberU64() < start {
+		// Return [] if the requested range is past our latest block
+		return []*beacon.ExecutionPayloadBodyV1{}
+	}
+	bodies := make([]*beacon.ExecutionPayloadBodyV1, count)
+	for i := uint64(0); i < count; i++ {
+		block := api.eth.BlockChain().GetBlockByNumber(start + i)
+		if block == nil {
+			continue
+		}
+		bodies[i] = getBody(block)
```

Comment:
You can move the `block==nil`-check inside `getBody`, and make these loops even smaller